op-chain-ops/cmd/op-migrate/main.go

```diff
@@ -0,0 +1,310 @@
+			dbHandles := ctx.Int("db-handles")
+			dbPath := ctx.String("db-path")
+			log.Info("Opening database", "dbCache", dbCache, "dbHandles", dbHandles, "dbPath", dbPath)
+			ldb, err := Open(dbPath, dbCache, dbHandles)
+			if err != nil {
+				return fmt.Errorf("cannot open DB: %w", err)
+			}
+
+			dryRun := ctx.Bool("dry-run")
+			noCheck := ctx.Bool("no-check")
+			if noCheck {
+				panic("must run with check on")
+			}
+
+			// Perform the migration
+			_, err = MigrateDB(ldb, !dryRun, noCheck)
+			if err != nil {
+				return err
+			}
+
+			// Close the database handle
+			if err := ldb.Close(); err != nil {
+				return err
+			}
+
+			log.Info("Finished migration successfully!")
+
+			return nil
+		},
+	}
+
+	if err := app.Run(os.Args); err != nil {
+		log.Crit("error in migration", "err", err)
+	}
+}
+
+type MigrationResult struct {
+	TransitionHeight    uint64
+	TransitionTimestamp uint64
+	TransitionBlockHash common.Hash
+}
+
+// MigrateDB will migrate an celo database to a new OP genesis block
+func MigrateDB(ldb ethdb.Database, commit, noCheck bool) (*MigrationResult, error) {
+	log.Info("Migrating DB")
+
+	// Grab the hash of the tip of the legacy chain.
+	hash := rawdb.ReadHeadHeaderHash(ldb)
+	log.Info("Reading chain tip from database", "hash", hash)
+
+	// Grab the header number.
+	num := rawdb.ReadHeaderNumber(ldb, hash)
+	if num == nil {
+		return nil, fmt.Errorf("cannot find header number for %s", hash)
+	}
+	log.Info("Reading chain tip num from database", "number", num)
+
+	// Grab the full header.
+	header := rawdb.ReadHeader(ldb, hash, *num)
+	trieRoot := header.Root
+	log.Info("Read header from database", "number", header)
+
+	// We need to update the chain config to set the correct hardforks.
+	genesisHash := rawdb.ReadCanonicalHash(ldb, 0)
+	cfg := rawdb.ReadChainConfig(ldb, genesisHash)
+	if cfg == nil {
+		log.Crit("chain config not found")
+	}
+	log.Info("Read config from database", "config", cfg)
+
+	// dbFactory := func() (*state.StateDB, error) {
+	// 	// Set up the backing store.
+	// 	underlyingDB := state.NewDatabaseWithConfig(ldb, &trie.Config{
+	// 		Preimages: true,
+	// 		Cache:     1024,
+	// 	})
+
+	// 	// Open up the state database.
+	// 	db, err := state.New(header.Root, underlyingDB, nil)
+	// 	if err != nil {
+	// 		return nil, fmt.Errorf("cannot open StateDB: %w", err)
+	// 	}
+
+	// 	return db, nil
+	// }
+
+	// db, err := dbFactory()
+	// if err != nil {
+	// 	return nil, fmt.Errorf("cannot create StateDB: %w", err)
+	// }
+
+	// Remove old blocks, so that we start with a fresh genesis block
+	currentHash := header.ParentHash
+	for {
+		// There are no uncles in Celo
+		num = rawdb.ReadHeaderNumber(ldb, currentHash)
+		hash = rawdb.ReadCanonicalHash(ldb, *num)
+		// if header == nil {
+		// 	return nil, fmt.Errorf("couldn't find header")
+		// }
+
+		log.Info("Deleting block", "hash", currentHash, "c", hash, "number", *num)
+		// rawdb.DeleteBlock(ldb, currentHash, *num)
```

Comment:
Didn't we want to delete all old blocks? If so, why is this commented out?

Reply:
Nice catch, this is a leftover from debugging. I'll wrap a check for `dru-run` around it.